src/js/strategies/CitrixVDIStrategy.js

```diff
@@ -6,28 +6,41 @@ import CCPInitiationStrategyInterface from "./CCPInitiationStrategyInterface";
 import {hitch} from "../utils";
 import {FailedState} from "../rtc_session";
 import {RTC_ERRORS} from "../rtc_const";
-import "@citrix/ucsdk/CitrixWebRTC";
+
+
 
 export default class CitrixVDIStrategy extends CCPInitiationStrategyInterface {
 
-    constructor() {
+    constructor(useRealCitrix = true) {
         super();
-        if (!window.CitrixWebRTC.isFeatureOn("webrtc1.0")) {
-            throw new Error('Citrix WebRTC redirection feature is NOT supported!');
-        }
-        window.getCitrixWebrtcRedir = function () {
-            const registryValue = Promise.resolve(1);
-            return new Promise(function (resolve, reject) {
-                //retrieve registry value internally
-                registryValue.then((v) => {
-                    resolve(v);
-                }).catch(() => {
-                    reject();
-                })
-            })
+        if(useRealCitrix){
+            require("@citrix/ucsdk/CitrixWebRTC");
         }
+        console.log("CitrixVDIStrategy initializing");
+        this.initCitrixWebRTC();
+        this.initGetCitrixWebrtcRedir();
+        this.initLog();
+    }
+
+    initCitrixWebRTC() {
+        window.CitrixWebRTC.setVMEventCallback((event) => {
+            if (event.event === 'vdiClientConnected') {
+                if (!window.CitrixWebRTC.isFeatureOn("webrtc1.0")) {
+                    throw new Error('Citrix WebRTC redirection feature is NOT supported!');
+                }
+                console.log("CitrixVDIStrategy initialized");
+            } else if (event.event === 'vdiClientDisconnected') {
+                console.log("vdiClientDisconnected");
```

Comment:
[Query] Do we need to throw exception if VDI client is disconnected?
Maybe we could log this using CCP logger as well.

Reply:
There could be cases where the disconnection was done gracefully. So exception might not always be the right choice here?

Reply:
based on the documentation, my understanding is this disconnected would happen only if the env setup is wrong, or agent disconnect from Citrix VDI. As we are not seeing any missed call or dropped call due to this issue, I assume we can leave this as a potential improvement item and see if we need to handle it based on the agent experience